src/Authentication/Authenticators/JWT/Firebase.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace CodeIgniter\Shield\Authentication\Authenticators\JWT;
+
+use CodeIgniter\Shield\Exceptions\RuntimeException;
+use DomainException;
+use Firebase\JWT\BeforeValidException;
+use Firebase\JWT\ExpiredException;
+use Firebase\JWT\JWT;
+use Firebase\JWT\Key;
+use Firebase\JWT\SignatureInvalidException;
+use InvalidArgumentException;
+use stdClass;
+use UnexpectedValueException;
+
+class Firebase implements JWTDecoderInterface
+{
+    /**
+     * Decode JWT
+     *
+     * @return stdClass Payload
+     */
+    public static function decode(string $encodedToken): stdClass
+    {
+        $config = setting('Auth.jwtConfig');
+
+        $key       = $config['secretKey'];
+        $algorithm = $config['algorithm'];
+
+        try {
+            return JWT::decode($encodedToken, new Key($key, $algorithm));
+        } catch (BeforeValidException|ExpiredException $e) {
+            throw new RuntimeException('Expired JWT: ' . $e->getMessage(), 0, $e);
+        } catch (
+            InvalidArgumentException|DomainException|UnexpectedValueException
+            |SignatureInvalidException $e
```

Comment:
I don't think these are valid syntax for 7.4, right? Could do one broader catch or cascading statements.

Reply:
> As of PHP 7.1.0, a [catch](https://www.php.net/manual/en/language.exceptions.php#language.exceptions.catch) block may specify multiple exceptions using the pipe (|) character. This is useful for when different exceptions from different class hierarchies are handled the same.
https://www.php.net/manual/en/language.exceptions.php#language.exceptions.catch